main.c

```diff
@@ -559,6 +559,11 @@ int main(int argc, char** argv)
         }
     }
 
+    if (flag_type >= FLAG_TYPE_END) {
```

Comment:
Add check for lower values as well
```suggestion
    if (flag_type <= FLAG_TYPE_INVALID || flag_type >= FLAG_TYPE_END) {
```